### DISCUSSION TITLE
Finder hangs when accessing mounted Google Drive folder

### DISCUSSION THREAD

**Discussion by manuel-koch** [iterate-ch/cyberduck#16481] [Q&A]
**Describe the bug**
Trying to access "My Drive" from my Google Drive ( which used to work ) but now whenever I actually try to access any of the files / folders in MacOS Finder it freezes.
I can see the file/folder structure of my Google Drive in Finder, but trying to access one of the items makes Finder hang/unresponsive.

Trying to disconnect that Google Drive via MountainDuck context menu does not unmount it, it appears frozen too.
Need to kill Finder and MountainDuck to make whole system working/responsive again.

**To Reproduce**
1. Opening MountainDuck menu
2. Click on 'Google Drive'
3. Click on "Verbinden"
4. Click on "Im Finder anzeigen"
5. Access a file in Finder ( of that mounted Google Drive )
6. Finder hangs/unresponsive, can't access anything 

**Expected behavior**
Accessing files/folders in mounted Google Drive in MacOS Finder works as expected.

**Screenshots**
na

**Desktop (please complete the following information):**
 - OS: MacOS Sonoma 14.7.1 (23H222) Apple M3 Pro
- MountainDuck: Version 4.16.3 (22374)

Unmounting/ejecting the "Google Drive" from Finder results in error message like
<img width="301" alt="Bildschirmfoto 2024-11-02 um 18 11 48" src="https://github.com/user-attachments/assets/887b69ea-7257-49eb-8cc3-569569f537db">

I can access my Google Drive with CyberDuck ! Only MountainDuck seem to have problems with Google Drive.

**Comment by manuel-koch** [comment]
Don't know why but after trying the following I can access my Google Drive again in MacOS Finder:
- disabled sychronization via menu "Synchronisierung anhalten"
- disconnected the mounted Google Drive via menu "Trennen"
- enabled additional logs via setting "Erweitertes Log aktivieren"
- connected/Mounted Google Drive again via menu "Verbinden"
- enabled sychronization via menu "Synchronisierung fortsetzen"
- Accessing files/folders in my Google Drive works as normal

**Reply by tatze96** [reply]
Thanks for letting us know. Please send us the logging output with activated debug logging (erweitertes Log) to support@mountainduck.io for further investigation if the issue re-appears.

**Comment by manuel-koch** [comment]
After upgrade to MacOS Sequoia 15.1 (24B83) the same behavior of hanging/unresponsive Finder when accessing files from my mounted Google Drive appeared on another computer.

**Comment by manuel-koch** [comment]
While Finder hangs and is unresponsive I can see 100% CPU load of `mds_stores` and `Mountain Duck` process.

**Reply by dkocher** [reply]
That would indicate Spotlight is indexing the mounted volume. Can you check the mount location in _Preferences… → Sync → Mount Location_ is set to the default `~/Library/Application Support/Mountain Duck/Volumes.noindex`. You can alternatively reset it in Terminal.app using 

```
defaults delete io.mountainduck mount.directory
```

**Reply by manuel-koch** [reply]
My setting already has the value `~/Library/Application Support/Mountain Duck/Volumes.noindex`.

**Comment by manuel-koch** [comment]
Maybe unrelated, but the problem vanished after I skipped ( or delayed execution ) a configured MacOS logon script that will load all my identities into ssh-agent.
One of the identities is password protected, thus the console script will wait until a password is entered:

`add_ssh_keys.sh`
```bash
#!/bin/bash
# Add known ssh keys
#
for k in aaa bbb ccc ddd ; do
    ssh-add ~/.ssh/$k
done
```

<img width="816" alt="Bildschirmfoto 2024-11-10 um 18 24 44" src="https://github.com/user-attachments/assets/072c87aa-9851-4a32-b2b1-4fd3caa58342">

Thus the strange behavior is as follows
- MacOS starts
- Configured logon action is triggered, shows a dialog whether user wants to start the script
- (A) Accepting the popup dialog, logon action is executed
- `add_ssh_keys.sh` starts running loading identities into ssh-agent
- the script stops when processing the password protected identity, i.e. `ssh-add` shows input prompt in terminal and waits/blocks for user input
- Mountain Duck starts, mounts my Google Drive
- I start browsing my mounted Google Drive in Finder
- **BOOM** Finder and Mountain Duck hang, are unresponsive

If I skip accepting the logon action (A) or just delay running it **until after** Mountain Duck started and mounted my Google Drive everything works as expected and I can access the files of my Google Drive.

Looks like ssh-agent is in some way blocking Mounting Duck to get information of the other SFTP related drives that I have configured, not necessarily my Google Drive as it obviously doesn't require any SSH identities.

@tatze96 Is this something worth investigating ?

**Reply by dkocher** [reply]
It is plausible that another connection configured in Mountain Duck that uses SFTP is blocked when trying to obtain the identities from the SSH agent because of the script. However from your description it sounds like a bug that this would also affect other connections you have configured such as Google Drive.
